### PR TITLE
feat: anchor links for params and return types

### PR DESCRIPTION
**Pull request by mxschmitt**
~~I thought about doing it via swizzling again the Docusaurus components, there I could have my own implementation for a list item. But I miss there a lot of information so I then tried it directly via the generator.~~

~~It seems to work the only thing where I was stuck was to build a recursive string of the parameters. e.g. `browser.newPage` has `geolocation.latitude` and in my current implementation it results in `#browser-new-page-param-latitude`. The ideal value would be `#browser-new-page-param-geolocation-latitude`~~

### CODE REVIEW

**Comment by pavelfeldman** [comment]
> The ideal value would be #browser-new-page-param-geolocation-latitude

I would say ideal would be The ideal value would be #browser-new-page-param-geolocation. We don't need to link deeper.

File: src/css/custom.css [hunk 1]
```diff
@@ -52,3 +52,19 @@ html[data-theme='dark'] .header-github-link:before {
...
+  position: relative;
+}
+
+a.list-anchor {
+  padding-left: 5px;
+  opacity: 0;
+  transition: opacity var(--ifm-transition-fast);
+}
+
+a.list-anchor:hover {
```
Comment (pavelfeldman) [inline src/css/custom.css:68]: I would do `li:hover > a.list-anchor:hover {`

It gives a bit noise via highlighting containers too, but I think it is ok. We can always follow up with a sprinkle of JS that puts "hovered" class only on the innermost hovered li and then it'll be ideal.

Comment (mxschmitt) [reply src/css/custom.css:68]: I like it, more accessible an behaves more like the normal heading anchor tag.

**Comment by pavelfeldman** [comment]
Looks like there still is a problem:
https://deploy-preview-240--playwright.netlify.app/docs/next/api/class-page#page-expose-binding-option-name
https://deploy-preview-240--playwright.netlify.app/docs/next/api/class-page#page-expose-binding-option-handle

These ^^ can't be on the same level, otherwise they can clash. Should be:

https://deploy-preview-240--playwright.netlify.app/docs/next/api/class-page#page-expose-binding-param-name
https://deploy-preview-240--playwright.netlify.app/docs/next/api/class-page#page-expose-binding-option-handle

File: src/generator.js [hunk 2]
```diff
@@ -471,4 +482,21 @@ function calculateHeadingHash(member) {
...
+ * @returns {String}
+ */
+function calculatePropertyHash(member, parentMember, direction) {
+  const className = toKebabCase(member.enclosingMethod.clazz.name);
+  const memberName = toKebabCase(member.enclosingMethod.name);
+  const prefix = `${className}-${memberName}`;
+  if (direction === 'out')
+    return `${prefix}-return`;
+  const propertyName = toKebabCase(member.name);
+  const propertyDescription = parentMember?.name === 'options' ? 'option': 'param';
```
Comment (pavelfeldman) [inline src/generator.js:498]: JavaScript: `#page-add-script-tag-option-content`
Python: `#page-add-script-tag-param-content`

File: src/documentation.js [hunk 3]
```diff
@@ -304,6 +304,8 @@ Documentation.Member = class {
     };
     this.async = false;
     this.alias = name;
+    /** @type {'param' | 'option'} */
+    this.argType = null;
```
Comment (pavelfeldman) [inline src/documentation.js:308]: `this.type` vs `this.argType` is confusing. i'd call it differently, `this.paramOrOption` :)

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**